### PR TITLE
Iris 4555 let the user edit a group of contacts

### PR DESCRIPTION
**Pull request by CataldoMazzilli**

### CODE REVIEW

File: src/network/api/modify-contact.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+				{ n: 'nickname', _content: name }
+		  ]
+		: undefined;
+	return modifyContact({ id, addedMembers, removedMembers, attributes }).then(
+		(res: ModifyContactResponse) =>
+			({
+				id: res.cn[0].id,
+				title: res.cn[0]._attrs.fullName ?? '',
+				members: res.cn[0].m?.map((value) => value.value) ?? []
+			} as ContactGroup)
```
Comment (beawar) [inline src/network/api/modify-contact.ts]: Why the cast?

File: src/views/board/common-contact-group-board.test.tsx [hunk 2]
```diff
@@ -0,0 +1,573 @@
...
+				});
+				expect(
+					screen.getByRoleWithIcon('button', { icon: TESTID_SELECTORS.icons.addMembers })
+				).toBeDisabled();
+			});
+
+			it('should enable the plus button when the user add a chip from the dropdown', async () => {
+				const email = faker.internet.email();
+				getSetupServer().use(
+					rest.post('/service/soap/FullAutocompleteRequest', async (req, res, ctx) =>
```
Comment (beawar) [inline src/views/board/common-contact-group-board.test.tsx]: the manual handler can be replaced with the registerFullAutocompleteHandler

**Review by giuliano176 — DISMISSED** [review]

**Review by FrancescoGottardi — DISMISSED** [review]

**Review by giuliano176 — DISMISSED** [review]

File: src/tests/utils.ts [hunk 3]
```diff
@@ -82,3 +83,27 @@ export async function delayUntil(emitter: EventEmitter, event: string): Promise<
 		emitter.once(event, resolve);
 	});
 }
+
+export const createCnItem = (
+	contactGroupName = faker.company.name(),
+	members: string[] = [],
+	id = faker.number.int({ min: 100 }).toString()
+): CnItem => {
+	const mappedMembers = members.map((member) => ({ type: 'I' as const, value: member }));
```
Comment (beawar) [inline src/tests/utils.ts]:
```suggestion
	const mappedMembers = members.map<CnItem['m'][number]>((member) => ({
		type: 'I',
		value: member
	}));
```

File: src/views/board/common-contact-group-board.test.tsx [hunk 4]
```diff
@@ -0,0 +1,562 @@
...
+		updateBoard: updateBoardFn ?? jest.fn(),
+		setCurrentBoard: jest.fn(),
+		getBoardContext: jest.fn(),
+		getBoard: jest.fn(),
+		closeBoard: closeBoardFn ?? jest.fn()
+	});
+}
+
+const contactGroup = buildContactGroup();
+beforeAll(() => {
```
Comment (beawar) [inline src/views/board/common-contact-group-board.test.tsx]: You probably want to use a beforeEach here, in order to have different mocks for each test

**Review by gnekoz — DISMISSED** [review]

File: src/views/board/edit-contact-group-board.test.tsx [hunk 5]
```diff
@@ -0,0 +1,648 @@
...
+		id: '',
+		url: '',
+		app: '',
+		icon: '',
+		title: ''
+	});
+}
+
+const contactGroup = buildContactGroup();
+beforeAll(() => {
```
Comment (beawar) [inline src/views/board/edit-contact-group-board.test.tsx]: beforeEach is safer

File: src/views/board/edit-contact-group-board.test.tsx [hunk 6]
```diff
@@ -0,0 +1,648 @@
...
+	spyUseBoardHooks();
+	spyUseBoard(contactGroup.id);
+	useContactGroupStore.getState().addStoredContactGroups([contactGroup]);
```
Comment (beawar) [inline src/views/board/edit-contact-group-board.test.tsx:50]: Reset the store before each test and then set the new state

**Review by beawar — APPROVED** [review]

**Review by FrancescoGottardi — APPROVED** [review]

**Review by giuliano176 — APPROVED** [review]

**Pull request merged**